ingest/downloader/downloader.py

```diff
@@ -0,0 +1,98 @@
+from typing import List
+
+from openpyxl import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+
+EXCLUDE_KEYS = ['describedBy', 'schema_type']
+
+
+class XlsDownloader:
+    def __init__(self):
+        self.workbook = {}
+        self.row = 1
+
+    def convert_json(self, entity_list: List[dict]):
+        self._flatten_object_list(entity_list)
+        return self.workbook
+
+    def _flatten_object_list(self, object_list: List[dict], object_key: str = ''):
+        for entity in object_list:
+            worksheet_name = object_key
+            row = {}
+            content = entity
+
+            if not object_key:
+                content = entity['content']
+                worksheet_name = self.get_concrete_entity(content)
+                row = {f'{worksheet_name}.uuid': entity['uuid']['uuid']}
+
+            if not worksheet_name:
+                raise Exception('There should be a worksheet name')
+
+            user_friendly_worksheet_name = self._format_worksheet_name(worksheet_name)
+
+            rows = self.workbook.get(user_friendly_worksheet_name, [])
+            self.workbook[user_friendly_worksheet_name] = rows
+            self._flatten_object(content, row, parent_key=worksheet_name)
+            rows.append(row)
+
+    def _flatten_object(self, object: dict, flattened_object: dict, parent_key: str = ''):
+        if isinstance(object, dict):
+            for key in object:
+                full_key = f'{parent_key}.{key}' if parent_key else key
+                if key in EXCLUDE_KEYS:
```

Comment:
Why is this line after 42?

Reply:
yeah, the full_key variable is needed later so I moved the if the condition on top.